modules/misc/src/postprocessors/ThermocoupleSensorPostprocessor.C

```diff
@@ -0,0 +1,302 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ThermocoupleSensorPostprocessor.h"
+#include "GeneralSensorPostprocessor.h"
+#include "Function.h"
+#include "MooseRandom.h"
+#include "LinearInterpolation.h"
+#include "SplineInterpolation.h"
+
+registerMooseObject("MiscApp", ThermocoupleSensorPostprocessor);
+
+InputParameters
+ThermocoupleSensorPostprocessor::validParams()
+{
+  InputParameters params = GeneralSensorPostprocessor::validParams();
+  params.addParam<string>("thermocouple_type", "B", "The type of thermocouple (B, E, J or K)");
+  params.addParam<string>("method", "numerical", "The equation to be used (lumped or numerical).");
+  params.addClassDescription("This is a ThermocoupleSensorPostprocessor for various classes of "
+                             "thermocouples, described by the 'thermocouple_type' parameter");
+  params.addParam<Real>("proportional_weight", 0, "The weight assigned to the proportional term");
+  params.addParam<Real>("integral_weight", 1, "The weight assigned to the integral term");
+  return params;
+}
+
+ThermocoupleSensorPostprocessor::ThermocoupleSensorPostprocessor(const InputParameters & parameters)
+  : GeneralSensorPostprocessor(parameters),
+    _thermocouple_type(getParam<string>("thermocouple_type")),
+    _method(getParam<string>("method"))
+{
+}
+
+void
+ThermocoupleSensorPostprocessor::initialize()
+{
+  // setting seed for random number generator
+  _rng.seed(_seed);
+  Real drift_value = 0.0;
+  Real efficiency_value = 0.0;
+  Real signalToNoise_value = 0.0;
+  Real noise_value = 0.0;
+  Real noise_std_dev = 0.0;
+  Real uncertainty_value = 0.0;
+  Real uncertainty_std_dev = 0.0;
+  Real delay_value = 0.0;
+
+  // setting values for thermocouple types B, J, E, K
+  // current values are placeholders
+  if (_thermocouple_type == "B")
+  {
+    // B type, temperature in degree celcius
+    if (_input_signal >= 0 && _input_signal < 630.615)
+    {
+      drift_value = 1.0;
+      efficiency_value = 1.0;
+      signalToNoise_value = 1.0;
+      noise_std_dev = 1.0;
```

Comment:
this means ignoring a lot of parameters the user could have set

you must error in the constructor if the user set these parameters and this is ignoring them

Reply:
we want the user to be able to change these values for thermocouples if they want because these will depend on the build of the instrument

Reply:
right now the user cannot though, since they are hard-coded to 1 in this mode,
if you use this mode, you must error when the user sets these parameters